src/utils/legend.coffee

```diff
@@ -65,63 +76,63 @@
+            item.append('circle')
+              .attr(
+                'fill-opacity': 0
+                'stroke': s.color
+                'stroke-width': '2px'
+                'r': d/2
+              )
+
+            item.append('text')
+              .attr(
+                'class': (d, i) -> "legendText series_#{i}"
+                'font-family': 'Courier'
+                'font-size': 10
+                'transform': 'translate(13, 4)'
+                'text-rendering': 'geometric-precision'
+              )
+              .text(s.label || s.y)
+
+        # Translate every legend g node to its position
+        translateLegends = () ->
+          [left, right] = that.computeLegendLayout(svg, series, dimensions)
+          groups
+            .attr(
+              'transform': (s, i) ->
+                if s.axis is 'y'
+                  return "translate(#{left.shift()},#{dimensions.height-40})"
+                else
+                  return "translate(#{right.shift()},#{dimensions.height-40})"
+            )
 
-        item.on('click', (s, i) ->
-          visibility = !(s.visible isnt false)
-          dispatch.toggle(s, i, visibility)
-          handlers.onSeriesVisibilityChange?({
-            series: s,
-            index: i,
-            newVisibility: visibility
-          })
-        )
-
-        item.append('circle')
-          .attr(
-            'fill': (s) -> s.color
-            'stroke': (s) -> s.color
-            'stroke-width': '2px'
-            'r': d/2
-          )
-
-        item.append('path')
-          .attr(
-            'clip-path': 'url(#legend-clip)'
-            'fill-opacity': (s) -> if s.type in ['area', 'column'] then '1' else '0'
-            'fill': 'white'
-            'stroke': 'white'
-            'stroke-width': '2px'
-            'd': (s) -> that.getLegendItemPath(s, d, d)
-          )
-
-        item.append('circle')
-          .attr(
-            'fill-opacity': 0
-            'stroke': (s) -> s.color
-            'stroke-width': '2px'
-            'r': d/2
-          )
-
-
-        item.append('text')
-          .attr(
-            'class': (d, i) -> "legendText series_#{i}"
-            'font-family': 'Courier'
-            'font-size': 10
-            'transform': 'translate(13, 4)'
-            'text-rendering': 'geometric-precision'
-          )
-          .text (s) -> s.label || s.y
-
+        # we need to do call this once, to not make the legend text
+        # does not blink on every update
+        translateLegends()
 
-        [left, right] = this.computeLegendLayout(svg, series, dimensions)
-        items.attr(
-          'transform': (s, i) ->
-            if s.axis is 'y'
-              return "translate(#{left.shift()},#{dimensions.height-40})"
-            else
-              return "translate(#{right.shift()},#{dimensions.height-40})"
-        )
+        # now once again,
+        # to make sure, text width gets really! computed properly
+        setTimeout translateLegends, 0
```

Comment:
Please use parenthesis for method calls (this is one of the CoffeeScript's features I don't like ^^)